pkg/cmd/ensureenv/ensureenv.go

```diff
@@ -0,0 +1,161 @@
+	fs.StringSliceVar(&e.shouldEqual, "ip-ev-equals", []string{}, "Validate that the value of the given environment variables are equal.")
+}
+
+func (e *ensureOpts) Validate() (err error) {
+	// Validate that the incoming environment variables are set and have non-empty values.
+	for _, ev := range e.notEmpty {
+		if value, has := os.LookupEnv(ev); !has {
+			return fmt.Errorf("The value of environment variable %v must be set.", ev)
+		} else if value == "" {
+			return fmt.Errorf("The value of environment variable %v must not be empty.", ev)
+		}
+	}
+
+	// Validate that the given environment variables are equal; this does not distinguish if the values are empty.
+	for _, ev := range e.shouldEqual {
+		var (
+			ipEV = os.Getenv(e.ipEV)
+			v    = os.Getenv(ev)
+		)
+		// Check both the raw value and the value as an array (wrapped in []).
+		//		If they aren't equal and we accept an invalid ip-ev value, don't error and continue on.
+		if !e.allowInvalidIPEV && (v != ipEV && v != fmt.Sprintf("[%s]", ipEV)) {
+			return fmt.Errorf("The value of environment variables %v, %v must be equal. Were: \"%v\", \"%v\"", e.ipEV, ev, ipEV, v)
+		}
+	}
+
+	// If --allow-invalid-ip-ev-val is true, --fallback-ip-ev MUST be set and have a non-empty value.
+	if e.allowInvalidIPEV {
+		if e.fallbackIPEV == "" {
+			return fmt.Errorf("Since --allow-invalid-ip-ev-val is true, --fallback-ip-ev must be provided.")
+		} else if os.Getenv(e.fallbackIPEV) == "" {
+			return fmt.Errorf("Since --allow-invalid-ip-ev-val is true, the value of the environment variable provided by --fallback-ip-ev [%v] must not be empty.", e.fallbackIPEV)
+		}
+	}
+
+	return
+}
+
+func (e *ensureOpts) Run() (err error) {
+	var (
+		fallbackEV = os.Getenv(e.fallbackIPEV)
+		ipEV       = os.Getenv(e.ipEV)
+	)
+
+	// If we have a value for ip-ev and it must be equal to
+	if ipEV == "" {
+		var (
+			interfaces []net.Interface
+		)
+		// Since the value of ip-ev is empty, we're going to check our
+		//   fallback environment variable against our network
+		//   interfaces, to see if we find a match.
+		// If we find a match, we'll consider ourselves valid, if we can't
+		//   we'll fail.
+		if interfaces, err = net.Interfaces(); err != nil {
+			return
+		}
+
+		for _, iface := range interfaces {
+			var (
+				addrs []net.Addr
+			)
+
+			if addrs, err = iface.Addrs(); err != nil {
+				return
+			}
+
+			for _, addr := range addrs {
+				if strings.Contains(addr.String(), fallbackEV) {
+					// We found a match, we'll assume now that the value of fallbackEV is good
+					//		and we'll continue allowing etcd to start.
+					return nil
+				}
+			}
+		}
+
+		// If we drop out of the loop, we haven't found a match and we need to error-out
+		return fmt.Errorf("Failed to find any ip addresses in network interfaces that match the value of %v", e.fallbackIPEV)
```

Comment:
style nits:

- error message does not start with capital letter.
- "failed to find ip address on network interfaces: %v"